docs/DeveloperGuide.md

```diff
@@ -1,34 +1,422 @@
+2. The module specified by the user is already added -- No module will be added.
+3. The module specified by the user is provided by NUS and have not been added -- Module will be added.
+
+#### Feature Implementation
+
+This feature is facilitated by the `AddModuleCommand` class which add the modules specified by the user. It overrides the `execute()` method which extends from the abstract `AddCommand` class which extends from the abstract `Command` class.   The `execute()` method's role is to execute the adding module operation and do necessary checks .
+
+The `AddCommand` will first try to call the static method `add` in `ModuleManager`  class, which will try to add the module specified by the user, exception will thrown according the following rules:
+
+1. `DuplicateModuleException` will be thrown if the module specified by the user is contained in the `ArrayList` named `moduleList` in `ModuleManager` class.
+2. `ModuleNotProvidedException` will be thrown if the module code specified by the user is not contained in the `HashMap` named `modulesMap` in `ModuleManager` class.
+
+Below are the class-diagram for the involved classes:
+
+![image-20200326014336120](images/Add_Module_Command_Class_Diagram.png)
+
+<span style="color: green"><small><i>Figure <b>Add Module Command Class Diagram</b></i></small></span>
+
+#### Example Usage
+
+The addition of modules will be illustrated as follows.
+
+James is a user and wants to add the module with the module code of "CS3235". Assume that he has the current modules:
+
+```
++--------------------------------------------------------------------------------------------------+
+ NO |  MODULE CODE   |                                 MODULE TITLE
++--------------------------------------------------------------------------------------------------+
+ 1  |     CS1231     |                             Discrete Structures
+ 2  |     CS2100     |                            Computer Organisation
+ 3  |     CS2113     |              Software Engineering & Object-Oriented Programming
++--------------------------------------------------------------------------------------------------+
+Total modules: 3
++--------------------------------------------------------------------------------------------------+
+```
+
+1. James will simply enter the command `addm cs3235`
+
+   After the input is parsed as an **add module task** and executed, the `AddModuleCommand#execute()` will call `ModuleManager#add()` to add the module cs3235. In the `ModuleManager#add()` method, it will call `ModuleManager#contains()` to check if the module cs3235 exists in the `ArrayList` named `moduleList` , then it will check if the module code "cs3235" is a key in the `HashMap` named `modulesMap`, after all, it will instantiate an `Module` object with the module code "CS3235" and respective title "Computer Security", then add the object into the `moduleList`.
+
+2. James receive the following feedback:
+
+   ```
+   root :
```

Comment:
I think that a picture would be better as it show what the actual output should look like 🤔